src/main/scala/com/gu/newsletterlistcleanse/sqs/AwsSQSSend.scala

```diff
@@ -1,23 +1,30 @@
 package com.gu.newsletterlistcleanse.sqs
 
 import com.amazonaws.regions.Regions
-import com.amazonaws.services.sqs.model.SendMessageRequest
+import com.amazonaws.services.sqs.model.{SendMessageBatchRequest, SendMessageBatchRequestEntry, SendMessageBatchResult, SendMessageRequest, SendMessageResult}
 import com.amazonaws.services.sqs.{AmazonSQSAsync, AmazonSQSAsyncClientBuilder}
 import com.gu.newsletterlistcleanse.NewsletterSQSAWSCredentialProvider
 import org.slf4j.{Logger, LoggerFactory}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-import scala.util.{Failure, Success, Try}
+import collection.JavaConverters._
+import scala.util.{Failure, Success}
+
 
 
 object AwsSQSSend {
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  def apply(queueName: QueueName)(payload: Payload):Future[Unit] = {
+  def apply[P <: Payload](queueName: QueueName)(payload : P):Future[Unit] = {
```

Comment:
personally I wouldn't bother with this and just get people to call either sendSingle or sendBatch explicitly.

Also, since I wrote the original AwsSQSSend it has turned out calling methods `apply` is not obvious to non scala devs, so that's another reason to have explicitly named methods.

Reply:
Yeah, I've decided to move it out to named methods instead. It definitely makes it easier to read and understand if you're not a confident Scala dev